test/Microsoft.AspNet.Mvc.ModelBinding.Test/Metadata/ModelMetadataProviderTest.cs

```diff
@@ -696,11 +696,75 @@ public void IsRequired_DataMemberIsRequiredTrueWithoutDataContract_False()
             Assert.False(metadata.IsRequired);
         }
 
+        [Fact]
+        public void PropertySetter_NotNull_ForPublicSetProperty()
+        {
+            // Arrange
+            var metadataProvider = TestModelMetadataProvider.CreateDefaultProvider();
+
+            // Act
+            var metadata = metadataProvider.GetMetadataForProperty(
+                typeof(ClassWithPublicSetProperty),
+                nameof(ClassWithPublicSetProperty.StringProperty));
+
+            // Assert
+            Assert.NotNull(metadata.PropertySetter);
+            Assert.NotNull(metadata.PropertyAccessor);
+        }
+
+        [Fact]
+        public void PropertySetter_Null_ForPrivateSetProperty()
+        {
+            // Arrange
+            var metadataProvider = TestModelMetadataProvider.CreateDefaultProvider();
+
+            // Act
+            var metadata = metadataProvider.GetMetadataForProperty(
+                typeof(ClassWithPrivateSetProperty),
+                nameof(ClassWithPrivateSetProperty.StringProperty));
+
+            // Assert
+            Assert.Null(metadata.PropertySetter);
+            Assert.NotNull(metadata.PropertyAccessor);
+        }
+
+        [Fact]
+        public void Metadata_Null_ForPrivateGetProperty()
+        {
+            // Arrange
+            var metadataProvider = TestModelMetadataProvider.CreateDefaultProvider();
+            var type = typeof(ClassWithPrivateGetProperty);
+            var propertyName = nameof(ClassWithPrivateGetProperty.StringProperty);
+
+            // Act & Assert
+            var exception = Assert.Throws<ArgumentException>(() =>
+            {
+                metadataProvider.GetMetadataForProperty(type, propertyName);
+            });
+            Assert.Equal($"The property {type}.{propertyName} could not be found.{Environment.NewLine}Parameter name: propertyName",
+                exception.Message);
```

Comment:
Just do `GetMetadataForType` and then check that the property is null, will be clearer.